### PR TITLE
Implement user settings page #18

### PR DESCRIPTION
**Pull request by PiaSophieBe**

### CODE REVIEW

**Comment by Melodycel** [comment]
I would like to do the review.

**Review by Melodycel — CHANGES_REQUESTED** [review]
It looks really good! 
I tested it and everything seems to work. The tests passed as well.
I just have some questions and notes.
Why did you choose these countries? Perhaps you could integrate more European countries such as Italy, Poland, Denmark or some German-speaking countries such as Austria and Switzerland. 
You say that the data of the account will be irrevocably deleted. Have you tested this? 
Maybe you could translate "Registered Date" and "Last Login Date" under „Meine Daten“ into German (Datum der Registrierung/ Letzter Login).

**Comment by astanik** [comment]
@Melodycel Thank you for your comment. Can you please use the github review functionality? ...
grafik


And then mark the code snippets you are talking about ...
grafik

File: src/views/AccountSettingsView.vue [hunk 1]
```diff
@@ -1,36 +1,272 @@
...
 export default {
   data() {
     return {
-      userEmail: ""
-    }
+      userProfile: {} as User, // Das gesamte Benutzerprofil
+      editedUserProfile: {} as User,
+      editMode: false, // Modus für die Bearbeitung der Benutzerdaten
+      visible: false, // Sichtbarkeit des Dialogs für Konto löschen
+      countries: [
```
[the remark this answers is not in the document]
Comment (Melodycel) [reply src/views/AccountSettingsView.vue:19]: Looks good!

File: src/views/AccountSettingsView.vue [hunk 2]
```diff
@@ -1,36 +1,272 @@
...
+    </div>
+    <div v-if="userProfile">
+      <div v-if="!editMode">
+        <div class='buttons-container centered-buttons'>
+          <Button type="button" icon="pi pi-user-edit" class='edit-button' @click="toggleEditMode" label="Bearbeiten" />
+          <Button type="button" icon="pi pi-trash" severity="danger" aria-label="Cancel" class='edit-button' label="Konto löschen" @click="visible = true" />
+        </div>
+        <div class='delete Button'>
+          <Dialog v-model:visible="visible" maximizable modal header=" " :style="{ width: '50rem' }" :breakpoints="{ '1199px': '75vw', '575px': '90vw' }">
+            <p>Bist du sicher, dass du dein Konto löschen möchtest? Alle deine Daten werden unwiderruflich gelöscht.</p>
```
[the remark this answers is not in the document]
Comment (Melodycel) [reply src/views/AccountSettingsView.vue:280]: After oral consultation: It was tested.

File: src/views/AccountSettingsView.vue [hunk 3]
```diff
@@ -1,36 +1,272 @@
...
+              <p v-if="userProfile.firstName || userProfile.lastName"><strong>Name:</strong> {{ userProfile.firstName }}
+                {{ userProfile.lastName }}</p>
+              <p v-if="userProfile.businessPhoneNumber"><strong>Geschäftliche Telefonnummer:</strong>
+                {{ userProfile.businessPhoneNumber }}</p>
+              <p v-if="userProfile.mobilePhoneNumber"><strong>Handynummer:</strong> {{ userProfile.mobilePhoneNumber }}
+              </p>
+              <p v-if="userProfile.privatePhoneNumber"><strong>Private Telefonnummer:</strong>
+                {{ userProfile.privatePhoneNumber }}</p>
+              <p v-if="userProfile.email"><strong>Email:</strong> {{ userProfile.email }}</p>
+              <p v-if="userProfile.registeredDate"><strong>Registered Date:</strong> {{ userProfile.registeredDate }}
```
[the remark this answers is not in the document]
Comment (Melodycel) [reply src/views/AccountSettingsView.vue]: Looks good!

**Review by Melodycel — APPROVED** [review]

**Review by Melodycel — APPROVED** [review]
Everything looks really good!

**Pull request merged**